client/src/components/ConnectToRoom.vue

```diff
@@ -109,9 +134,64 @@ export default {
             this.createAnswer();
           });
       });
+
       this.$socket.$subscribe("title", title => {
         this.$store.dispatch("UPDATE_STREAM_TITLE", title);
       });
+
+
+      /* Listeners to convert this client into host for new peers */
+
+
+      this.$socket.$subscribe("peer joined", peerData => {
+        if(peerData.hostID !== this.$socket.client.id) {
+          console.log("peer not for me");
+          return;
+        }
+
+
+        this.peers[peerData.id] = {
+          id: peerData.id,
+          room: peerData.room,
+          iceCandidates: []
+        };
+
+        startShare(peerData.id, this);
+      });
+
+      this.$socket.$subscribe("peer ice", iceData => {
+        console.log("Ice Candidate from peer: " + iceData.id + " in room: " + iceData.room);
+        console.log("Ice Candidate: " + iceData.candidate);
+        if (this.$store.getters.ROOM != iceData.room ||
```

Comment:
This is quite a complex if statement. What does it logically check for, and could we rip this out into a function?

Reply:
When the client receives "peer ice" event in the socket room, it's only meant for one client in the room whose socket id == hostID sent by the peer. This checks if the ice data received is for itself. I don't think there's a need to create a separate function just for an if condition

Reply:
I think I agree. A comment like "Checks if ICE candidate from peer is meant for this socket" could be useful.